### PR TITLE
[Doc] add explanations for subscription mode

### PR DESCRIPTION
**Pull request by Anonymitaet**
- This PR add docs for https://github.com/apache/pulsar/pull/10817

- This PR updates `master` firstly. Will update other doc versions once the changes get approved.

- Preview looks good:
image
image
image

### CODE REVIEW

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -396,6 +397,57 @@ producer = client.create_producer(topic='my-topic', batching_type=pulsar.Batchin
...
 > * You cannot use cumulative acknowledgment with Key_Shared type.
 
+### Subscription mode
+
+#### What is subscription mode
+
+The subscription mode indicates the cursor type. 
+
+- When a subscription is created, an associated cursor is created to record the last consumed position. 
+
```
Comment (BewareMyPower) [inline site2/docs/concepts-messaging.md:407]: I think no blank line is needed for a list in markdown.

```
- line 01
- line 02
```

File: site2/docs/admin-api-topics.md [hunk 2]
```diff
@@ -1464,3 +1440,90 @@ If a message has a key, it supersedes the round robin routing policy. The follow
...
+<!--Java-->
+
+```java
+String topic = "persistent://my-tenant/my-namespace/my-topic";
+admin.topics().getSubscriptions(topic);
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+### Delete (unsubscribe) subscription
```
Comment (BewareMyPower) [inline site2/docs/admin-api-topics.md:1503]: Just use `Delete subscription` or keep the original `Unsubscribe` as the title.

File: site2/docs/admin-api-topics.md [hunk 3]
```diff
@@ -1464,3 +1440,90 @@ If a message has a key, it supersedes the round robin routing policy. The follow
...
+When a subscription does not process messages any more, you can delete (unsubscribe) using one of the following methods. 
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--pulsar-admin-->
+
+```shell
+pulsar-admin topics unsubscribe \
+--subscription my-subscription \
+persistent://test-tenant/ns1/tp1 \
```
Comment (BewareMyPower) [inline site2/docs/admin-api-topics.md:1514]:
```suggestion
persistent://test-tenant/ns1/tp1
```

`\` is to separated multiple lines in a shell command. But it should not appear at the tail

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: site2/docs/concepts-messaging.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:407]: Yes, but I insert a blank line to make things more clear.

File: site2/docs/admin-api-topics.md (hunk 2 above)
Comment (Anonymitaet) [reply site2/docs/admin-api-topics.md]: OK, I prefer `Unsubscribe`. I've updated the PR, PTAL. @BewareMyPower

File: site2/docs/admin-api-topics.md (hunk 3 above)
Comment (Anonymitaet) [reply site2/docs/admin-api-topics.md]: Oops, yes. Updated.

**Review by BewareMyPower — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Comment by Anonymitaet** [comment]
Hi @sijia-w I've added explanations for the subscription mode in this PR. Can you help make the same changes to the following doc versions? Thanks

- 2.7.0，2.7.1，2.7.2，2.7.3，2.8.0，2.8.1，2.8.2 in `website/versioned_docs` folder

- 2.7.1, 2.7.2, 2.7.3, 2.8.0 in `website-next/docs` folder

File: site2/docs/concepts-messaging.md [hunk 4]
```diff
@@ -316,10 +316,11 @@ A subscription is a named configuration rule that determines how messages are de
 > * If you want to achieve "message queuing" among consumers, share the same subscription name among multiple consumers(shared, failover, key_shared).
 > * If you want to achieve both effects simultaneously, combine exclusive subscription type with other subscription types for consumers.
 
-### Consumerless Subscriptions and Their Corresponding Types
+### Subscriptions type
```
Comment (sijia-w) [inline site2/docs/concepts-messaging.md:319]: Shall we use `Subscription types` instead?

File: site2/docs/concepts-messaging.md [hunk 5]
```diff
@@ -396,6 +397,57 @@ producer = client.create_producer(topic='my-topic', batching_type=pulsar.Batchin
 > * You need to specify a key or orderingKey for messages.
 > * You cannot use cumulative acknowledgment with Key_Shared type.
 
+### Subscription mode
```
Comment (sijia-w) [inline site2/docs/concepts-messaging.md:400]: This heading should be in the same format of `Subscription types`. I suggest we use `Subscription modes`.

File: site2/docs/concepts-messaging.md [hunk 6]
```diff
@@ -396,6 +397,57 @@ producer = client.create_producer(topic='my-topic', batching_type=pulsar.Batchin
...
+
+#### What is subscription mode
```
Comment (sijia-w) [inline site2/docs/concepts-messaging.md:402]: Here, we should either use `a subscription mode` or `subscription modes`. @Anonymitaet

File: site2/docs/concepts-messaging.md (hunk 4 above)
Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:319]: Ah, yes, thanks for reminding me, can you update it in your PR https://github.com/apache/pulsar/pull/12794? thanks

File: site2/docs/concepts-messaging.md (hunk 5 above)
Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:400]: Subscription modes

File: site2/docs/concepts-messaging.md (hunk 6 above)
Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:402]: `subscription modes`

can you update it in your PR #12794? thanks

Comment (sijia-w) [reply site2/docs/concepts-messaging.md:402]: Done.

**Pull request merged**